screensaver/screensaver.py

```diff
@@ -74,86 +93,269 @@ def get_last_level_directories(photo_dirs_list) -> list:
+    return size
 
-    delete_old_screensaver_photos()
 
-    # Now run rsync
+def is_remote_available():
     proc = subprocess.run(["rsync",
-                           # verbose
-                           "-v",
-                           # recursive
-                           "-r",
-                           # protect arguments --protect-args equivalent
-                           "-s",
+                           # verbose + dry-run
+                           "-vn",
                            # use special port
                            "--rsh", "ssh -p" + RSYNC_PORT,
-                           # include files:
-                           "--include", "*.jpg",
-                           "--include", "*.JPG",
-                           "--include", "*.jpeg",
-                           "--include", "*.JPEG",
+                           # include directories"
+                           "--include", "*/",
                            # exclude files:
                            "--exclude", "*",
                            # from:
-                           f"{INPUT_PATH}{random_dir_choice}/",
+                           INPUT_PATH,
                            # to:
-                           OUTPUT_PATH],
-                          stdout=subprocess.PIPE,  # required to return names of photos transferred
-                          universal_newlines=True  # required to return string, not bytes-like obj
-                          # stderr=subprocess.STDOUT
+                           "."
+                           ],
+                          stdout=subprocess.PIPE,
+                          stderr=subprocess.PIPE
                           )
+    output_err = str(proc.stderr)
+    # print(output_err)
+    if "Could not resolve hostname" in output_err:
+        return False
+    elif "Permission denied" in output_err:
+        return False
+    elif "Connection closed by" in output_err:
+        return False
+    elif "connection unexpectedly closed" in output_err:
+        return False
+    else:
+        return True
+
+
+def get_random_entry(dirs_list) -> str:
+    return np.random.choice(dirs_list)
+
+
+def setup_paths(photos_path, library_path):
+    # Delete existing photos
+    delete_directory(photos_path)
+    # Make new photos folder
+    make_directory(photos_path)
+    subprocess.run(["chmod", "777", photos_path])
+
+    if not os.path.exists(library_path):
+        make_directory(library_path)
+        subprocess.run(["chmod", "777", library_path])
+    pass
 
-    # Change mode of newly created directory so photos can be deleted when job runs
-    subprocess.run(["chmod", "777", OUTPUT_PATH])
 
-    # Lower case output to handle cases when file extension is in caps
-    proc_stdout_lower_case = proc.stdout.lower()
+def get_already_used_list():
+    # Load list of directories that have already been shown
+    # If this is the first time the script runs it will start with an empty one
```

Comment:
These comments should be in a doc string. Also function annotation is missing again.